### PR TITLE
Add precommit format

### PR DESCRIPTION
**Pull request by justinjmoses**
This will fail the commit if the lint fails, but the formatted files are not added to the commit. 

This could be fixed by adding a `git add -u` but then that may unintentionally add files to the commit that aren't intended. This should really be amended to "format everything staged and stage those formats". Looking into options if anyone has any good suggestions?

### CODE REVIEW

**Review by zyzek — DISMISSED** [review]

**Review by hav-noms — DISMISSED** [review]
You asked us to review even tho still in Draft. Cant wait for this to speed up PR's with all those pesky np's!

File: package.json [hunk 1]
```diff
@@ -55,6 +58,11 @@
 	"directories": {
 		"test": "test"
 	},
+	"husky": {
+		"hooks": {
+			"pre-commit": "npm run lint && npm run format"
```
Comment (jacko125) [inline package.json:63]: this would be super helpful

**Review by jacko125 — DISMISSED** [review]

**Comment by justinjmoses** [comment]
Note: I updated this to use `pretty-quick` to do the formatting as well

**Review by zyzek — DISMISSED** [review]

**Review by jacko125 — APPROVED** [review]
LGTM, installed husky locally already

**Review by ajsantander — DISMISSED** [review]
Awesome!

**Pull request merged**